### PR TITLE
improve renegotiation ssl tests

### PR DESCRIPTION
**Pull request by wfurt**
Improve test to make sure we actually do renegotiation. 
It is curious that we actually never set _pendingReHandshake to true in production code. 

I will address it as part of #453

### CODE REVIEW

**Comment by wfurt** [comment]
Outerloop test failures are unrelated. Updated renegotiate tests passed.

**Review by eiriktsarpalis — APPROVED** [review]
LGTM 👍

File: src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamAllowRenegotiationTests.cs [hunk 1]
```diff
@@ -52,8 +60,8 @@ public async Task SslStream_AllowRenegotiation_True_Succeeds()
                 // Initiate Read operation, that results in starting renegotiation as per server response to the above request.
                 int bytesRead = await ssl.ReadAsync(message, 0, message.Length);
 
-                // There's no good way to ensure renegotiation happened in the test.
-                // Under the debugger, we can see this test hits the renegotiation codepath.
+                // Renegotiation will trigger another validation callback/
+                Assert.True(validationCount > 1);
```
Comment (stephentoub) [inline src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamAllowRenegotiationTests.cs]:
```suggestion
                Assert.InRange(validationCount, 2, int.MaxValue);
```

File: src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs [hunk 2]
```diff
@@ -152,8 +160,9 @@ public async Task SslStream_NetworkStream_Renegotiation_Succeeds()
                 // Initiate Read operation, that results in starting renegotiation as per server response to the above request.
                 int bytesRead = await ssl.ReadAsync(message, 0, message.Length);
 
-                // There's no good way to ensure renegotiation happened in the test.
-                // Under the debugger, we can see this test hits the renegotiation codepath.
+                // renegotiation will trigger validation callback again.
+                Assert.True(validationCount > 1);
```
Comment (stephentoub) [inline src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamNetworkStreamTest.cs]:
```suggestion
                Assert.InRange(validationCount, 2, int.MaxValue);
```

**Review by stephentoub — APPROVED** [review]

**Review by davidsh — APPROVED** [review]

**Pull request merged**